File: Data Structures & Algorithms/sum-of-all-subset-xor-totals/submission-0.java
class Solution {
    public int subsetXORSum(int[] nums) {
        return backtrack(nums,  0 , 0);
    }
        private int backtrack(int[] nums, int i, int curr_xor){
            if(i == nums.length){
                return curr_xor;}
            
            int include = backtrack(nums, i+1, curr_xor ^ nums[i]);
            int exclude = backtrack(nums, i+1, curr_xor);

            return include + exclude;

            }   
    }

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:

        n = len(nums)
        res = []
        visited = set()
        def backtrack(curr_arr ):
            if len(curr_arr) == n:
                res.append(curr_arr[:])
                return

            for num in nums:
                if num not in visited:
                    visited.add(num)
                    curr_arr.append(num)
                    backtrack(curr_arr)
                    curr_arr.pop()
                    visited.remove(num)

        backtrack([])
        return res
            
        